deeptools/bamPEFragmentSize.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+
+# own tools
+from deeptools.getFragmentAndReadSize import get_read_and_fragment_length
+from deeptools._version import __version__
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description='Given a BAM file, several regions are'
+        'sampled to estimate the paired-end '
+        'fragment length.')
+    parser.add_argument('bam',
+                        help='BAM file to process',
+                        metavar='bam-file')
+
+    parser.add_argument('-bai',
+                        help='Index for the BAM file. Default is to '
+                        'consider the path of the BAM file adding '
+                        'the .bai suffix.',
+                        metavar='bam-file-index')
+
+    parser.add_argument('--histogram', '-hist',
+                        help='Save a .png file with a histogram '
+                        'of the fragment length distribution.',
+                        metavar='FILE')
+
+    parser.add_argument('--numberOfProcessors', '-p',
+                        help='Number of processors to use. The default is '
+                        'to use 1.',
+                        metavar="INT",
+                        type=int,
+                        default=1,
+                        required=False)
+
+    parser.add_argument('--verbose',
+                        help='Set if processing data messages are wanted.',
+                        action='store_true',
+                        required=False)
+
+    parser.add_argument('--version', action='version',
+                        version='%(prog)s {}'.format(__version__))
+
+    return parser
+
+
+def main(args=None):
+    args = parse_arguments().parse_args(args)
+    fragment_len_dict, read_len_dict = get_read_and_fragment_length(args.bam, args.bai, return_lengths=True,
+                                                                    numberOfProcessors=args.numberOfProcessors)
+
+    if fragment_len_dict['mean'] == 0:
+        print "No pairs were found. Is the dataset paired-end?"
+
+    print "Sample size: {}\n".format(fragment_len_dict['sample_size'])
+
+    print "\nFragment lengths:"
+    print "Min.: {}\n1st Qu.: {}\nMean: {}\nMedian: {}\n" \
+          "3rd Qu.: {}\nMax.: {}\nStd: {}".format(fragment_len_dict['min'],
+                                                  fragment_len_dict['qtile25'],
+                                                  fragment_len_dict['mean'],
+                                                  fragment_len_dict['median'],
+                                                  fragment_len_dict['qtile75'],
+                                                  fragment_len_dict['max'],
+                                                  fragment_len_dict['std'])
+
+    print "\nRead lengths:"
```

Comment:
Why would logging improve in this case? Do you need the output in certain way for galaxy?